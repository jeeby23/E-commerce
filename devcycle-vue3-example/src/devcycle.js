import * as DevCycle from '@devcycle/js-client-sdk'

let devcycleClient = null

export function initializeDevCycle() {
  const DEVCYCLE_CLIENT_SDK_KEY = import.meta.env.VITE_DEVCYCLE_CLIENT_SDK_KEY
  if (!DEVCYCLE_CLIENT_SDK_KEY) {
    alert('Set your VITE_DEVCYCLE_CLIENT_SDK_KEY environment variable to use the DevCycle JS SDK.')
  }

  devcycleClient = DevCycle.initializeDevCycle(
    DEVCYCLE_CLIENT_SDK_KEY,
    // Modify this user object to see how it affects targeting
    {
      user_id: 'user-1',
      name: 'User 1',
      email: 'user1@email.com'
    },
    { 
      logLevel: 'debug',
      // Controls the interval between flushing events to the DevCycle servers
      eventFlushIntervalMS: 1000
    }
  )
  return devcycleClient
}

export function getDevCycleClient() {
  return devcycleClient
}