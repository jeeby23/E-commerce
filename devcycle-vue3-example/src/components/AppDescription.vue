<template>
  <div className="App-description">
    <div v-if="text === 'step-1'">
      <h3>Welcome to DevCycle's example app.</h3>
      <p>
        If you got here through the onboarding flow, just follow the instructions to change and create new Variations and see how the app reacts to new Variable values.
      </p>
    </div>
    <div v-else-if="text === 'step-2'">
      <h3>Great! You've taken the first step in exploring DevCycle.</h3>
      <p>
        You've successfully toggled your very first Variation. You are now serving a different value to your users and you can see how the example app has reacted to this change.
      </p>
      <p>
        Next, go ahead and create a whole new Variation to see what else is possible in this app.
      </p>
    </div>
    <div v-else-if="text === 'step-3'">
      <h3>You're getting the hang of things.</h3>
      <p>
        By creating a new Variation with new Variable values and toggling it on for all users, you've already explored the fundamental concepts within DevCycle.
      </p>
      <p>
        There's still so much more to the platform, so go ahead and complete the onboarding flow and play around with the feature that controls this example in your dashboard.
      </p>
    </div>
    <div v-else>
      <h3>Welcome to DevCycle's example app.</h3>
      <p>
        If you got to the example app on your own, follow our README guide to create the Feature and Variables you need to control this app in DevCycle.
      </p>
    </div>
  </div>
</template>

<script>
import { getDevCycleClient } from '../devcycle.js';

export default {
  data() {
    return {
      text: 'default'
    }
  },
  mounted() {
    const devcycleClient = getDevCycleClient()
    this.text = devcycleClient.variableValue('example-text', 'default')
    devcycleClient.subscribe('variableUpdated:example-text', (key, updatedVariable) => {
      this.text = updatedVariable.value
    })
  }
}
</script>
