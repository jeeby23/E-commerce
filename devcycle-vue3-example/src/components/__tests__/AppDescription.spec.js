import { describe, test, expect, vi, beforeEach } from 'vitest'
import { mount, flushPromises } from '@vue/test-utils';

import { getDevCycleClient } from '../../devcycle.js';
import AppDescription from '../AppDescription.vue';

vi.mock('../../devcycle.js')

describe('AppDescription', () => {
    let mockGetDevCycleClient
    beforeEach(() => {
        mockGetDevCycleClient = vi.mocked(getDevCycleClient)
    })

    test.each([
        'default',
        'step-1',
        'step-2',
        'step-3',
    ])('renders the app description correctly for value "%s"', async (variableValue) => {
        mockGetDevCycleClient.mockReturnValue({
            variableValue: vi.fn().mockReturnValue(variableValue),
            subscribe: vi.fn()
        })
        const wrapper = mount(AppDescription);
        await flushPromises();
        expect(wrapper.text()).toMatchSnapshot()
    });
});
