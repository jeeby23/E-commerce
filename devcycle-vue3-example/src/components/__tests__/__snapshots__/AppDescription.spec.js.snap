// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppDescription > renders the app description correctly for value "default" 1`] = `"Welcome to DevCycle's example app. If you got to the example app on your own, follow our README guide to create the Feature and Variables you need to control this app in DevCycle."`;

exports[`AppDescription > renders the app description correctly for value "step-1" 1`] = `"Welcome to DevCycle's example app. If you got here through the onboarding flow, just follow the instructions to change and create new Variations and see how the app reacts to new Variable values."`;

exports[`AppDescription > renders the app description correctly for value "step-2" 1`] = `"Great! You've taken the first step in exploring DevCycle. You've successfully toggled your very first Variation. You are now serving a different value to your users and you can see how the example app has reacted to this change.  Next, go ahead and create a whole new Variation to see what else is possible in this app."`;

exports[`AppDescription > renders the app description correctly for value "step-3" 1`] = `"You're getting the hang of things. By creating a new Variation with new Variable values and toggling it on for all users, you've already explored the fundamental concepts within DevCycle.  There's still so much more to the platform, so go ahead and complete the onboarding flow and play around with the feature that controls this example in your dashboard."`;
